ewatercycle/models/marrmot.py

```diff
@@ -173,7 +215,7 @@ def _create_marrmot_config(self) -> PathLike:
             store_ini=self.store_ini,
         )
 
-        config_file = self.work_dir / 'marrmot-m01_config.mat'
+        config_file = work_dir / Path('marrmot-m01_config.mat')
```

Comment:
```suggestion
        config_file = work_dir / 'marrmot-m01_config.mat'
```
if `work_dir` is a Path instance (as the type hints suggests), then this should just work I think.

Reply:
First I applied your suggestion, but then I bring it back due to mypy error:  `Unsupported left operand type for / ("_PathLike[Any]")`

Reply:
That's strange. Can you try calling this (on the command line with you ewatercycle conda env activated)?

`mypy -c 'from pathlib import Path; Path(__file__).parent.parent / "foo" / "bar"'`

(from: https://gitter.im/python/typing?at=5d25171ff867ec1687ddc61a)

Reply:
Your command line does not return an error. It seems that there is a difference in using `from pathlib import Path` vs  `from os import PathLike`. You can try the lines below in a python script and run the mypy on that script. It reproduces the error that I mentioned. 

```python
from pathlib import Path
from os import PathLike

def pathlike_fun(work_dir: PathLike):
    config_file = work_dir / 'marrmot-m01_config.mat'
    return config_file

def path_fun(work_dir: Path):
    config_file = work_dir / 'marrmot-m01_config.mat'
    return config_file
```

Reply:
Yep, then I also get the error. @sverhoeven what is the reason to use `PathLike` rather than `Path` or perhaps `PurePath`?

Reply:
No particular reason, it was used in the rest of the code so I continued with it.

Reply:
Haha, ok I did the same. Perhaps we can change it here @SarahAlidoost ?

Reply:
I couldn't change it only in this part of the code, because mypy returned several `Incompatible types in assignment` errors in other parts. So, I skipped changing it for now.